core/amber/src/main/scala/edu/uci/ics/texera/web/resource/aiassistant/AiAssistantResource.scala

```diff
@@ -10,4 +13,42 @@ class AiAssistantResource {
   @RolesAllowed(Array("REGULAR", "ADMIN"))
   @Path("/isenabled")
   def isAiAssistantEnable: String = isEnabled
+
+  /**
+    * To get the type annotation suggestion from OpenAI
+    */
+  @POST
+  @RolesAllowed(Array("REGULAR", "ADMIN"))
+  @Path("/getresult")
+  def getAiResponse(prompt: String, @Auth user: SessionUser): Response = {
+    val finalPrompt = prompt.replace("\\", "\\\\").replace("\"", "\\\"")
+    val requestBody =
+      s"""
+         |{
+         |  "model": "gpt-4",
+         |  "messages": [{"role": "user", "content": "$finalPrompt"}],
+         |  "max_tokens": 15
+         |}
+            """.stripMargin
+
+    try {
+      val url = new java.net.URL(s"${AiAssistantManager.sharedUrl}/chat/completions")
+      val connection = url.openConnection().asInstanceOf[java.net.HttpURLConnection]
+      connection.setRequestMethod("POST")
+      connection.setRequestProperty("Authorization", s"Bearer ${AiAssistantManager.accountKey}")
+      connection.setRequestProperty("Content-Type", "application/json")
+      connection.setDoOutput(true)
```

Comment:
this too low level... to avoid reimplmenet wheels, I suggest use some https request libraries. I am not sure which one we are using currently. but some example can be https://github.com/scalaj/scalaj-http.

Reply:
I check the build.sbt and we are using `import kong.unirest.Unirest`.

Reply:
sounds good please use that one.